src/gridcoin/beacon.cpp

```diff
@@ -62,7 +65,12 @@ bool TryRenewal(BeaconRegistry::BeaconMap& beacons, const BeaconPayload& payload
         return false;
     }
 
+    // Insert current beacon into historical map.
+    historical_beacons.emplace(std::make_pair(ctx_hash, current_beacon));
```

Comment:
I haven't dug into where this is used, but don't you want to insert the historical beacon after the timestamp and hash are assigned?

Reply:
See the latest push....